api/api_ethereum.go

```diff
@@ -502,52 +502,220 @@ type EthRPCTransaction struct {
+		Type:     typeInt,
+		From:     getFrom(tx),
+		Gas:      hexutil.Uint64(tx.Gas()),
+		GasPrice: (*hexutil.Big)(tx.GasPrice()),
+		Hash:     tx.Hash(),
+		Input:    tx.Data(),
+		Nonce:    hexutil.Uint64(tx.Nonce()),
+		To:       resolveToField(tx),
+		Value:    (*hexutil.Big)(tx.Value()),
+		V:        (*hexutil.Big)(signature.V),
+		R:        (*hexutil.Big)(signature.R),
+		S:        (*hexutil.Big)(signature.S),
+	}
+
+	if blockHash != (common.Hash{}) {
+		result.BlockHash = &blockHash
+		result.BlockNumber = (*hexutil.Big)(new(big.Int).SetUint64(blockNumber))
+		result.TransactionIndex = (*hexutil.Uint64)(&index)
+	}
+
+	// TODO-Klaytn: Have to add additional fields for ethereum transaction types.
+
+	return result
+}
+
+// newEthRPCPendingTransaction creates an EthRPCTransaction for pending tx.
+func newEthRPCPendingTransaction(tx *types.Transaction) *EthRPCTransaction {
+	return newEthRPCTransaction(tx, common.Hash{}, 0, 0)
+}
+
 // GetTransactionByBlockNumberAndIndex returns the transaction for the given block number and index.
 func (api *EthereumAPI) GetTransactionByBlockNumberAndIndex(ctx context.Context, blockNr rpc.BlockNumber, index hexutil.Uint) *EthRPCTransaction {
-	// TODO-Klaytn: Not implemented yet.
-	return nil
+	block, err := api.publicTransactionPoolAPI.b.BlockByNumber(ctx, blockNr)
+	if block == nil || err != nil {
+		return nil
+	}
+
+	ethTx := newEthRPCTransactionFromBlockIndex(block, uint64(index))
+	if ethTx == nil {
+		return nil
+	}
+
+	return ethTx
 }
 
 // GetTransactionByBlockHashAndIndex returns the transaction for the given block hash and index.
 func (api *EthereumAPI) GetTransactionByBlockHashAndIndex(ctx context.Context, blockHash common.Hash, index hexutil.Uint) *EthRPCTransaction {
-	// TODO-Klaytn: Not implemented yet.
+	block, err := api.publicTransactionPoolAPI.b.BlockByHash(ctx, blockHash)
+	if block != nil && err == nil {
+		ethTx := newEthRPCTransactionFromBlockIndex(block, uint64(index))
+		if ethTx == nil {
+			return nil
+		}
+		return ethTx
+	}
 	return nil
 }
 
 // GetRawTransactionByBlockNumberAndIndex returns the bytes of the transaction for the given block number and index.
 func (api *EthereumAPI) GetRawTransactionByBlockNumberAndIndex(ctx context.Context, blockNr rpc.BlockNumber, index hexutil.Uint) hexutil.Bytes {
-	// TODO-Klaytn: Not implemented yet.
-	return nil
+	rawTx, err := api.publicTransactionPoolAPI.GetRawTransactionByBlockNumberAndIndex(ctx, blockNr, index)
+	if err != nil {
+		return nil
+	}
+
+	return rawTx
 }
 
 // GetRawTransactionByBlockHashAndIndex returns the bytes of the transaction for the given block hash and index.
 func (api *EthereumAPI) GetRawTransactionByBlockHashAndIndex(ctx context.Context, blockHash common.Hash, index hexutil.Uint) hexutil.Bytes {
-	// TODO-Klaytn: Not implemented yet.
-	return nil
+	rawTx, err := api.publicTransactionPoolAPI.GetRawTransactionByBlockHashAndIndex(ctx, blockHash, index)
+	if rawTx == nil || err != nil {
```

Comment:
It is minor, but how about this for consistency?
```suggestion
	if err != nil {
```

Reply:
Actually there are similar checking codes in this PR.
So I'd like to hear original proposer @jimni1222 's opinion about it :)

For me, it looks good to change checking logic like @jeongkyun-oh commented.
If @jimni1222 is ok about it, I'll update this PR. 👍 

[GetTransactionByBlockNumberAndIndex](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR580)
```go
	block, err := api.publicTransactionPoolAPI.b.BlockByNumber(ctx, blockNr)
	if block == nil || err != nil {
		return nil
	}
```

[GetRawTransactionByBlockHashAndIndex](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR618) which is what @jeongkyun-oh commented.
```go
	rawTx, err := api.publicTransactionPoolAPI.GetRawTransactionByBlockHashAndIndex(ctx, blockHash, index)
	if rawTx == nil || err != nil {
		return nil
	}
```

[GetRawTransactionByHash](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR647)
```go
	rawTx, err := api.publicTransactionPoolAPI.GetRawTransactionByHash(ctx, hash)
	if rawTx == nil || err != nil {
		return nil, err
	}
```

[GetTransactionReceipt](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR657)
```go
	ethTx, err := newEthTransactionReceipt(tx, blockHash, blockNumber, index, cumulativeGasUsed, receipt)
	if ethTx == nil || err != nil {
		return nil, err
	}
```

[GetTransactionByBlockNumberAndIndex](https://github.com/klaytn/klaytn/pull/1093/files#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR580)
```go
	block, err := api.publicTransactionPoolAPI.b.BlockByNumber(ctx, blockNr)
	if block == nil || err != nil {
		return nil
	}
```

Reply:
If the implemented logic is not a part of the Ethereum logic as it is, you can freely change it.

Reply:
@jeongkyun-oh @jimni1222 
Applied the feedback [here](https://github.com/klaytn/klaytn/commit/558a0dda4ff27d46e80b1afad64b5c166bc4cfb2) :)